### PR TITLE
TST/BLD: unpin upper bluesky constraint

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Try releasing the upper pin on bluesky for future illumine compatibility

## Motivation and Context
Revisiting this now that numpy 2.0 turned our lives upside down

## How Has This Been Tested?
Go gha go

## Where Has This Been Documented?
This PR

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
bluesky only just updated for numpy 2.0 compatibility, there's a release notes PR up to tag those changes in v1.13.0a4.

**Comment by tangkong** [comment]
after some stupidity reporoducing this locally, a few notes:
* I've confirmed that latest bluesky removes all the numpy deprecation errors.  This version (1.13.0a4?) is not released yet
* Local test errors remaining are:
  * Recursion depth in test_optimize
  * unpickable _thread.lock
  * trying to call `.send()` on a list
* Recursion depth errors come from bluesky trying to deepcopy a `SynGauss` object that was stashed in the md-dictionary, not from the optimization routine itself

...investigation ongoing

**Comment by tangkong** [comment]
The `delay_scan` error I've encountered locally does not seem to appear in CI.  Currently CI only fails when trying to call `pack_event_page` with no arguments.  These go away once we upgrade past 1.12.0 (as can be seen in the conda builds)

I think this PR has correct changes, and the rest of the nabs test suite will have to wait until bluesky updates to 1.13.0a4

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ requirements:
   - pip
   run:
   - python >=3.6
-  - bluesky >=1.6.5,<1.11
+  - bluesky >=1.6.5
```
Comment (ZLLentz) [inline conda-recipe/meta.yaml:24]: Does this need to be re-pinned higher? Or maybe this is for the follow-up?

**Review by ZLLentz — APPROVED** [review]
I think these are all correct too

File: conda-recipe/meta.yaml (hunk 1 above)
Comment (tangkong) [reply conda-recipe/meta.yaml:24]: I'll leave this for the followup.  Older versions of bluesky would still work for numpy<2.0, technically.

**Comment by tangkong** [comment]
Though it hurts to bypass the tests, I'll ship this and we'll wait for the bluesky pin

**Pull request merged**